src/transaction_builder.js

```diff
@@ -190,16 +193,17 @@ function prepareInput (input, kpPubKey, redeemScript, hashType) {
 
       var prevOutScriptScriptHash = bscript.decompile(input.prevOutScript)[1]
       if (!bufferEquals(prevOutScriptScriptHash, redeemScriptHash)) throw new Error('Inconsistent hash160(RedeemScript)')
+    }
+
+    var expanded = expandOutput(redeemScript, undefined, kpPubKey)
+    if (!expanded.pubKeys) throw new Error('RedeemScript not supported "' + bscript.toASM(redeemScript) + '"')
 
-    // or, we don't have a prevOutScript, so generate a P2SH script
-    } else {
+    // if we don't have a prevOutScript, generate a P2SH script
+    if (!input.prevOutType) {
       input.prevOutScript = bscript.scriptHashOutput(redeemScriptHash)
```

Comment:
moved this to after the `throw` to avoid `input` mutation until we're on the home stretch... though, even then its not perfect,  but it is an aim